File: webapp/src/models/Order.ts
import { CartItem } from "./Cart";
import { IDish } from "./Dish";
import { IVatCategory } from "./VatCategory";
import { IDelivery } from "./Purchase";

export interface IOrder {
  id: number;
  orderNumber: string;
  orderDate: string;
  lowVatAmount: number;
  highVatAmount: number;
  totalAmount: number;
  deliveries: IDelivery[];
  transactions: ITransaction[];
  status: OrderStatus;
	lastName: string;
	firstName: string;
	street: string;
	houseNumber: number;
	addHouseNumber: string;
	zipcode: string;
	city: string;
	email: string;
	phone: string;
	details: string;
  dietdetails: string;
}

export interface IStatus {
  id: number,
  status: number
}

export enum OrderStatus {
  "Nieuw" = 0,
  "Betaling wordt verwerkt",
  "Betaling mislukt of afgebroken",
  "Betaling gelukt",
  "Verzonden",
  "Afgeleverd"
}

export interface ITransaction {
  id: number;
  orderId: number;
  transactionId: string;
  transactionStatus: string;
  transactionDateTime: string;
}


export class Order implements IOrder {
  id = 0;
  orderNumber = "";
  orderDate = "";
  lowVatAmount = 0;
  highVatAmount = 0;
  totalAmount = 0;
  deliveries = [];
  transactions = [];
  status = 0;
  lastName = "";
  firstName = "";
  street = "";
  houseNumber = 0;
  addHouseNumber = "";
  zipcode = "";
  city = "";
  email = "";
  phone = "";
  details = "";
  dietdetails = "";
}

File: WebApi/Data/DataObjects/Order.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CrossCuttingConcerns.EntityFrameworkGenerics;
using CrossCuttingConcerns.Enums;

namespace Data.DataObjects
{
   

    [Table("Order", Schema = "GoedEten")]
    public class Order : IEntity
    {
       
        public int Id { get; set; }
        public string OrderNumber { get; set; }
        [Required, Column(TypeName = "DateTime2(0)")]
        public DateTime OrderDate { get; set; }
        public decimal NoVatPrice { get; set; }
        public decimal LowVatPrice { get; set; }
        public decimal HighVatPrice { get; set; }
        public decimal TotalAmount { get; set; }
        public ICollection<OrderDelivery> Deliveries { get; set; }
        public ICollection<OrderTransaction> Transactions { get; set; }
        [Required]
        public OrderStatus Status { get; set; }
        //      public Address BillingAddress { get; set; }
        [Required]
        public string FirstName { get; set; }
        [Required]
        public string LastName { get; set; }
        [Required]
        public string Street { get; set; }
        [Required]
        public int HouseNumber { get; set; }
        public string AddHouseNumber { get; set; }
        [Required]
        public string Zipcode { get; set; }
        [Required]
        public string City { get; set; }
        [Required]
        public string Email { get; set; }
        [Required]
        public string Phone { get; set; }
        public string Details { get; set; }        
        public string Dietdetails { get; set; }

    }
    
    //[Owned]
    //public class Address
    //{     
        
    //    [Required]
    //    public string FirstName { get; set; }
    //    [Required]
    //    public string LastName { get; set; }
    //    [Required]
    //    public string Street { get; set; }        
    //    public int HouseNumber { get; set; }        
    //    public string AddHouseNumber { get; set; }
    //    [Required]
    //    public string Zipcode { get; set; }
    //    [Required]
    //    public string City { get; set; }

    //}
}

